### DISCUSSION TITLE
Some suggestion on display the content of long files

### DISCUSSION THREAD

**Discussion by Kralizek** [spectreconsole/spectre.console#1184] [Q&A]
I am to display the content of a long file.

```csharp
AnsiConsole.Write(new Text(content));
```

Simply reading the file and writing to the console works fine but I'd like to improve the user experience. I'm not even sure what kind of experience I want to give my user(s). I am thinking something like the ability to scroll up and down the content as it gets printed.

I'm curious to know if other developers have faced the same issue and what they came up with.

Thanks.

**Comment by rocklau** [comment]
or have a scrollable panel , I have a change some code for keep last message in panel
```csharp
        var allLines = Segment.SplitLines(childSegments, innerWidth, null).ToList();
        // If scrolling is enabled and content exceeds height limit, keep only the last few lines
        if (Scroll && height.HasValue && allLines.Count > height.Value)
        {
            allLines = allLines.Skip(allLines.Count - height.Value).ToList();
        }
```

https://github.com/rocklau/spectre.console/commit/df8aa875fd21b739c8e64abb0a35acdee169c316

**Comment by HeyItsGilbert** [comment]
I'm creating a live view where I have a panel that I'd like the user to be able to navigate. The content is regularly too long and it was would be awesome to render portions. I was thinking of adding a scrollPosition and then skipping anything less then that. I was running Render on the input objects to get an idea of how many lines there were but once I got there I couldn't figure out how to actually print things.
